gateway-server/src/main/java/org/apache/knox/gateway/websockets/GatewayWebsocketHandler.java

```diff
@@ -173,12 +174,41 @@ private synchronized String getMatchedBackendURL(final String path) {
 
     /* URL used to connect to websocket backend */
     String backendURL = urlFromServiceDefinition(serviceRegistryService, entry, path);
+    LOG.logMessage("Url obtained from services definition: " + backendURL);
 
     StringBuilder backend = new StringBuilder();
     try {
-
-      /* if we do not find websocket URL we default to HTTP */
-      if (!StringUtils.containsAny(backendURL, WEBSOCKET_PROTOCOL_STRING, SECURE_WEBSOCKET_PROTOCOL_STRING)) {
+      if (StringUtils.containsAny(backendURL, WEBSOCKET_PROTOCOL_STRING, SECURE_WEBSOCKET_PROTOCOL_STRING)) {
+        LOG.logMessage("ws or wss protocol found in service url");
+        URI serviceUri = new URI(backendURL);
+        backend.append(serviceUri);
+        /* Avoid Zeppelin Regression - as this would require ambari changes and break current knox websocket use case*/
+        if (!StringUtils.endsWith(backend.toString(), "/ws") && !StringUtils.endsWith(backend.toString(), "/websocket")
```

Comment:
Why are we adding special handling for `/wesocket` path ?

Reply:
Added by mistake while I was trying different things to fix issue. Removed now. Thanks